repository/repository-core/src/main/java/org/eclipse/vorto/repository/web/api/v1/NamespaceController.java

```diff
@@ -77,52 +408,342 @@
+
+  /**
+   * This is a temporary endpoint to create namespaces that will ultimately replace
+   * {@link TenantManagementController#createTenant(String, CreateTenantRequest)}. <br/>
+   * The endpoint takes an empty body and a namespace path variable, then tries to create it by
+   * associating the necessary user/tenant information before handing over to the autowired
+   * {@link ITenantService}.<br/>
+   * Everything "tenant" should be refactored and simplified further on, so the tenant service
+   * should be gone and replaced with a namespace service eventually.
+   * @param namespace
+   * @return
+   */
+  @RequestMapping(method = RequestMethod.PUT, value="/{namespace:.+}", produces = "application/json")
+  @PreAuthorize("isAuthenticated()")
+  public ResponseEntity<NamespaceOperationResult> createNamespace(
+      @ApiParam(value = "The name of the namespace to be created", required = true)
+      final @PathVariable String namespace
+  ) {
+    /*
+      This creates a fake "tenant ID" to feed the tenant service and pointlessly populate the
+      tenant table. Once everything "tenant-wise" is replaced with a simpler namespace-oriented
+      architecture, the UUID itself should be removed, i.e. a "namespace service" would not care
+      to use an additional unique ID since the namespace names are unique.
+      */
+    String fakeTenantId = UUID.randomUUID().toString().replace("-", "");
+    IUserContext userContext = UserContext.user(SecurityContextHolder.getContext().getAuthentication(), fakeTenantId);
+    // validating namespace notation and user-related (private vs public)
+    Optional<NamespaceOperationResult> validationError = NamespaceValidator.validate(namespace, userContext);
+    if (validationError.isPresent()) {
+      return new ResponseEntity<>(validationError.get(), HttpStatus.BAD_REQUEST);
+    }
+
+    try {
+      /*
+      This wraps the only admin (i.e. the given user for this request) in a set, as the service
+      API allows multiple admins (but it makes no sense upon creation of a new namespace - only
+      one admin and that is the user who initiated the request).
+       */
+      Set<String> fakeAdmins = new HashSet<>();
+      fakeAdmins.add(userContext.getUsername());
+      /*
+      This wraps the only namespace in a set, according to service API .
+       */
+      Set<String> fakeNamespaces = new HashSet<>();
+      fakeNamespaces.add(namespace);
+      tenantService.createOrUpdateTenant(
+          fakeTenantId,
+          namespace,
+          fakeAdmins,
+          Optional.of(fakeNamespaces),
+          // no authentication or authorization provider necessary
+          Optional.ofNullable(null),
+          Optional.ofNullable(null),
+          userContext
+      );
+
+      return new ResponseEntity<>(NamespaceOperationResult.success(), HttpStatus.OK);
+    }
+    catch (NamespaceExistException e) {
```

Comment:
Personal choice here, I like to give it some air so `else` and `catch` statements can have comments directly above. This going against formatting guidelines, but so does a significant part of the code base with no visible advantage. Not sure how strict we wanna be with that specifically to context?

Reply:
IMO it would be better to try to stick to the formatting guidelines, when we are changing or writing new code, otherwise the code will never conform to any formatting standard.

Reply:
Fair enough. I formatted the whole class according to the template. Still hate the inline else and catch blocks personally.